Fixed accidental trimming of spaces put into filters

jharmer95 opened the pull request:
Fixed issue ( #782 ) where trailing space was deleted automatically preventing the entering of filters that contain whitespace (e.g. a Category named "To Be Completed")

JosefNemec commented:
Have you tested that this doesn't break other places where this converter [is used](https://github.com/JosefNemec/Playnite/search?q=ListToStringConverter&unscoped_q=ListToStringConverter)? Or that the search still works properly when trailing white spaces are present? I'm particular skeptical about game edit window still working properly if you don't remove trailing spaces.

JosefNemec commented:
I don't have a problem with adding parameter controlling whether trimming should be used.

jharmer95 commented:
Added control logic for ListToStringConverter to determine whether a second trim occurs.
Added a ConverterParameter (string "NoTrim") to the Modern.xaml and Classic.xaml files to allow the filter fields to ignore final whitespace trimming, but the game edit and any other windows that use the ListToStringConverter won't be affected unless they are explicitly given a ConverterParameter of "NoTrim"

JosefNemec commented:
Sorry it took me song long to review this. As far as custom skins goes, that won't be an issue in future since theming API will be reworked and this will be handled differently.
